examples/jsm/exporters/GLTFExporter.js

```diff
@@ -1331,13 +1345,24 @@ class GLTFWriter {
 	 */
 	processTexture( map ) {
 
+		const writer = this;
+		const options = writer.options;
 		const cache = this.cache;
 		const json = this.json;
 
 		if ( cache.textures.has( map ) ) return cache.textures.get( map );
 
 		if ( ! json.textures ) json.textures = [];
 
+		// make non-readable textures (e.g. CompressedTexture) readable by blitting them into a new texture
+		if ( map instanceof CompressedTexture ) {
+
+			map = decompress( map, options.maxTextureSize );
+			// remove from cache, as the underlaying canvas is always the same between decompressed textures
+			cache.images.delete( map.image );
```

Comment:
map is new. However, map.image points to the same canvas. 

I think this is only relevant / correct for cases where a custom renderer would be passed into decompress, since the temporary `_renderer` and thus its canvas is now by default disposed each time. If you feel strongly about this I can remove it